x-pack/plugins/timelines/public/components/actions/timeline/cases/add_to_existing_case_button.tsx

```diff
@@ -25,21 +27,38 @@ const AddToCaseActionButtonComponent: React.FC<AddToCaseActionButtonProps> = ({
   owner,
   onClose,
 }) => {
-  const { addExistingCaseClick, isDisabled, userCanCrud } = useAddToCase({
+  const { onCaseSuccess, onCaseClicked, isDisabled, userCanCrud, caseAttachments } = useAddToCase({
     event,
     useInsertTimeline,
     casePermissions,
     appId,
     owner,
     onClose,
   });
+  const { cases } = useKibana<TimelinesStartServices>().services;
+  const addToCaseModal = cases.hooks.getUseCasesAddToExistingCaseModal({
+    attachments: caseAttachments,
+    updateCase: onCaseSuccess,
+    onRowClick: onCaseClicked,
+  });
+
+  // TODO To be further refactored and moved to cases plugins
+  // https://github.com/elastic/kibana/issues/123183
+  const handleClick = () => {
```

Comment:
This code is temporal. An upcoming PR will move this whole component out of timelines as well as the add to new case.